### ISSUE TITLE
Schema examples dont show

### ISSUE THREAD

**Issue by AngryUbuntuNerd** [sphinx-contrib/openapi#152]
I have the given [openapi.json](https://github.com/sphinx-contrib/openapi/files/14726482/openapi.json) which is generated using `fastapi` (I chopped it down to the relevant parts).

 The relevant snippet is this one:
```json
{
  "components": {
    "schemas": {
      "Schedule": {
        "examples": [
          {
            "schedule_id": "111111l901",
            "task": "ping"
          }
        ]
      },
      "ScheduleRequest": {
        "examples": [
          {
            "task": "ping"
          }
        ]
      }
    }
  }
}
```

This is following the [OpenAPI standard 3.1.0](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#schema-object) which claims:

> The `example` property has been deprecated in favor of the JSON Schema `examples` keyword. Use of `example` is discouraged, and later versions of this specification may remove it.

Now I have added it to my docs using this block
```
.. openapi:: ./openapi.json
   :examples:
```

But the generated examples are not using the `examples` from the document:

![Screenshot from 2024-03-22 17-40-01](https://github.com/sphinx-contrib/openapi/assets/3716309/b8ad13d9-0ac1-4bb9-affa-c58be0c897e3)
![Screenshot from 2024-03-22 17-39-55](https://github.com/sphinx-contrib/openapi/assets/3716309/2d616202-c6b7-44aa-82b0-a9e2c88efb1d)

Notice that it says `"string"` instead of `"111111l901"` and `"ping"` as in the `examples`.

I can not figure out how to fix this in the code, so was wondering if someone can help.

**Comment by stephenfin** [comment]
Yeah we don't support examples at the moment. Those would need to be added.

**Comment by AngryUbuntuNerd** [comment]
Happy to put together a PR for adding the support. Would just need a few pointers where to make the changes.
